File: server/Data/ProductData.cs
using System;
using uShopping.Models;

namespace uShopping.Data {
  public class ProductData {
    public Guid Id { get; set; }
    public Guid ListId { get; set; }
    public string Name { get; set; }
    public int? Count { get; set; }
   public bool? IsPurchased { get; set; }

    public ProductData () {}
    public ProductData (Product product) {
        ListId = product.ListId;
        Id = product.Id;
        Name = product.Name;
        Count = product.Count;
        IsPurchased = product.IsPurchased;
    }
  }
}

File: app/app/src/main/java/com/example/ushopping/data/ProductData.java
package com.example.ushopping.data;

import android.content.Intent;

import java.util.UUID;

public class ProductData {
    public UUID id;
    public UUID listId;
    public String name;
    public Integer count;
    public Boolean isPurchased;

    public ProductData () {}
    public ProductData(String name, Integer count) {
        this.name = name;
        this.count = count;
    }

    public ProductData(UUID listId, UUID id, boolean isPurchased) {
        this.listId = listId;
        this.id = id;
        this.isPurchased = isPurchased;
    }

}
